crates/pixi_config/src/gateway.rs

```diff
@@ -19,3 +22,19 @@ pub fn from_pixi_config(config: &Config) -> ChannelConfig {
         per_channel: Default::default(),
     }
 }
+
+/// Constructs a [`Gateway`] from a [`ClientWithMiddleware`] and a [`Config`]
+pub fn new_gateway(client: ClientWithMiddleware, config: Config) -> Gateway {
```

Comment:
we have a gateway per project, I thought it would be nice to have an API for some more global gateway